### PR TITLE
Update and add Cabinet archive related rules

### PR DESCRIPTION
**Pull request by jtothej**
Suggesting following rearrangement as FCI/FDI API functions are dependent on each other so I'm not sure if it's worth to have separate rules for each of them:

`FCICreate` and `FDICreate` can be potentially used for shellcode execution via callback functions (TODO) so adding them as library functions:
`lib/create-file-compression-interface-context.yml`
`lib/create-file-decompression-interface-context.yml`

Rules indicating creation or extraction of data from Cabinet file:
`data-manipulation/compression/create-cabinet-file.yml`
`data-manipulation/compression/extract-files-from-cabinet.yml`

CC: @mike-hunhoff

### CODE REVIEW

**Comment by mr-tz** [comment]
@mike-hunhoff, can you take a look?

**Comment by mr-tz** [comment]
@mike-hunhoff bump :)

File: data-manipulation/compression/create-cabinet-file.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+rule:
+  meta:
+    name: create Cabinet file
```
Comment (mike-hunhoff) [inline data-manipulation/compression/create-cabinet-file.yml:3]: Thoughts on changing this to `create Cabinet on Windows` to make it more clear?

File: data-manipulation/compression/create-cabinet-file.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - https://learn.microsoft.com/en-us/windows/win32/devnotes/creating-a-cabinet
+    examples:
+      - 44bad2e2a9e387b86870f009d01833ea4618d2a7cda5f64fa84a19f3bdf4efaf:0x1400028E0
+  features:
+    - and:
+      - match: create File Compression Interface context
+      - or:
+        - api: cabinet.FCIAddFile = add file to Cabinet
+        - api: cabinet.FCIFlushFolder = flush current folder under construction
+        - api: cabinet.FCIFlushCabinet = completes current cabinet
```
Comment (mike-hunhoff) [inline data-manipulation/compression/create-cabinet-file.yml:23]: Add `cabinet.FCIDestroy`?

File: data-manipulation/compression/extract-files-from-cabinet.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+rule:
+  meta:
+    name: extract files from Cabinet
```
Comment (mike-hunhoff) [inline data-manipulation/compression/extract-files-from-cabinet.yml:3]: Thoughts on changing this to `extract Cabinet on Windows` to make it more clear?

File: lib/create-file-compression-interface-context.yml [hunk 4]
```diff
@@ -1,12 +1,14 @@
 rule:
   meta:
-    name: open cabinet file
-    namespace: host-interaction/file-system
+    name: create File Compression Interface context
```
Comment (mike-hunhoff) [inline lib/create-file-compression-interface-context.yml:3]: Thoughts on adding `on Windows` to make it more clear?

File: lib/create-file-decompression-interface-context.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: create File Decompression Interface context
```
Comment (mike-hunhoff) [inline lib/create-file-decompression-interface-context.yml:3]: Thoughts on adding `on Windows` to make it more clear?

**Review by mike-hunhoff — CHANGES_REQUESTED** [review]
Thank you @jtothej ! I'm not sure why I missed all of the mentions here and I apologize for the delayed review. Please check out my comments.

File: data-manipulation/compression/create-cabinet-file.yml (hunk 1 above)
Comment (jtothej) [reply data-manipulation/compression/create-cabinet-file.yml]: Thanks! Renamed the rule in https://github.com/mandiant/capa-rules/pull/808/commits/ac09516cca968e54d6b5764405337cba3f290c21

File: data-manipulation/compression/extract-files-from-cabinet.yml (hunk 3 above)
Comment (jtothej) [reply data-manipulation/compression/extract-files-from-cabinet.yml]: Thanks! Renamed the rule in https://github.com/mandiant/capa-rules/pull/808/commits/ac09516cca968e54d6b5764405337cba3f290c21

File: lib/create-file-compression-interface-context.yml (hunk 4 above)
Comment (jtothej) [reply lib/create-file-compression-interface-context.yml]: Thanks! Renamed the rule in https://github.com/mandiant/capa-rules/pull/808/commits/ac09516cca968e54d6b5764405337cba3f290c21

File: lib/create-file-decompression-interface-context.yml (hunk 5 above)
Comment (jtothej) [reply lib/create-file-decompression-interface-context.yml]: Thanks! Renamed the rule in https://github.com/mandiant/capa-rules/pull/808/commits/ac09516cca968e54d6b5764405337cba3f290c21

File: data-manipulation/compression/create-cabinet-file.yml (hunk 2 above)
Comment (jtothej) [reply data-manipulation/compression/create-cabinet-file.yml]: Thanks! Added in https://github.com/mandiant/capa-rules/pull/808/commits/ac09516cca968e54d6b5764405337cba3f290c21

**Review by mr-tz — APPROVED** [review]
thanks, I think these look good now

**Pull request merged**